airbyte-server/src/main/java/io/airbyte/server/handlers/WorkspacesHandler.java

```diff
@@ -163,6 +164,11 @@ public WorkspaceRead getWorkspaceBySlug(final SlugRequestBody slugRequestBody)
     return buildWorkspaceRead(workspace);
   }
 
+  public WorkspaceRead getWorkspaceByConnectionId(final ConnectionIdRequestBody connectionIdRequestBody) {
+    final StandardWorkspace workspace = configRepository.getStandardWorkspaceFromConnection(connectionIdRequestBody.getConnectionId(), true);
```

Comment:
There was an existing method to get a workspace with a connection id, but I didn't understand the 2nd parameter. This is the method:

```
  public StandardWorkspace getStandardWorkspaceFromConnection(final UUID connectionId, final boolean isTombstone) {
    try {
      final StandardSync sync = getStandardSync(connectionId);
      final SourceConnection source = getSourceConnection(sync.getSourceId());
      return getStandardWorkspaceNoSecrets(source.getWorkspaceId(), isTombstone);
    } catch (final Exception e) {
      throw new RuntimeException(e);
    }
  }

  public StandardWorkspace getStandardWorkspaceNoSecrets(final UUID workspaceId, final boolean includeTombstone)
      throws JsonValidationException, IOException, ConfigNotFoundException {
    final StandardWorkspace workspace = persistence.getConfig(ConfigSchema.STANDARD_WORKSPACE, workspaceId.toString(), StandardWorkspace.class);

    if (!MoreBooleans.isTruthy(workspace.getTombstone()) || includeTombstone) {
      return workspace;
    }
    throw new ConfigNotFoundException(ConfigSchema.STANDARD_WORKSPACE, workspaceId.toString());
  }

```

I set includeTombstone to true, since that always returns the workspace, but I don't know what tombstone means here.

Reply:
I guess this is a way to not return workspaces that have been removed - I'm thinking of leaving this as true since we may want to use this more flexibly in the future, and then the user of the API can decide what to do with tombstoned workspaces

Reply:
Usually, tombstone would stand for records that have been deleted. `getStandardWorkspaceFromConnection` with `tombstone=true` means that we'd get deleted workspaces as well.
Also, looking at the `WorkspaceRead` definition, I don't think we expose the tombstone information.
I would lean towards not returning deleted objects by default because that feels more confusing from a caller's point of view, especially since there is no way to know that the object returned is in fact deleted.

Reply:
ah, didn't realize we didn't expose the tombstone data. makes sense to not return deleted ones then, will update